mmv1/templates/terraform/pre_create/alloydb_cluster.go.erb

```diff
@@ -30,4 +30,40 @@ if backupSource != nil || continuousBackupSource != nil {
 	}
 	restoreClusterRequestBody["cluster"] = cluster
 	obj = restoreClusterRequestBody
-}
\ No newline at end of file
+}
+
+
+// Read the secondary cluster config to call the api for creating secondary cluster
+
+var secondaryConfig interface{}
+var clusterType interface{}
+
+if val, ok := obj["secondaryConfig"]; ok {
+	secondaryConfig = val
+}
+
+if val, ok := obj["clusterType"]; ok {
```

Comment:
What if clusterType or secondaryConfig is not provided? They are not marked required in the .yaml above. I am assuming the cluster creation will proceed normally with the cluster type being primary.

Reply:
Yes., if both the fields 'clusterType' and 'secondaryConfig' are not provided. The default value is assumed to be PRIMARY and a primary cluster will be created

Moreover, If secondaryConfig is provided, then the clusterType has to be defined and should be set to SECONDARY. If not, the system will throw a validation error.

And If clusterType is set to SECONDARY then the secondaryConfig has to be defined or again a validation error is returned.